crates/ruma-client-api/src/session/login.rs

```diff
@@ -226,22 +226,33 @@ pub mod v3 {
     #[serde(tag = "type", rename = "m.login.password")]
     pub struct Password {
         /// Identification information for the user.
-        pub identifier: UserIdentifier,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub identifier: Option<UserIdentifier>,
 
         /// The password.
         pub password: String,
+
+        /// Username for the user.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[deprecated = "\
+            Since Matrix Client-Server API r0.4.0, clients should instead use \
+            identifiers (`UserIdentifier`) instead of user.\
+        "]
+        pub user: Option<String>,
     }
 
     impl Password {
         /// Creates a new `Password` with the given identifier and password.
-        pub fn new(identifier: UserIdentifier, password: String) -> Self {
-            Self { identifier, password }
+        #[allow(deprecated)]
+        pub fn new(identifier: Option<UserIdentifier>, password: String) -> Self {
```

Comment:
I believe we should keep `identifier` not an option in this constructor. Since `user` is deprecated, there is no good reason to construct this type without an identifier.

Reply:
Maybe you misunderstood me. `identifier` still needs to be an `Option` because it doesn't make sense to have this field alongside one of the deprecated fields. However we should have the constructor that take a `UserIdentifier` and sets `identifier: Some(identifier`.

Reply:
I think I get it now, is this good?